addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -98,10 +100,13 @@ public void importFromUrl(final String url) throws IllegalArgumentException, IOE
         }
 
         String collectionJson = requestor.getResponseBody(uri);
-        importCollection(collectionJson);
+        importCollection(collectionJson, variables);
     }
 
-    public void importCollection(String collection) throws JsonProcessingException {
+    public void importCollection(String collection, final String variables)
+            throws JsonProcessingException {
+        collection = replaceVariables(collection, variables);
```

Comment:
Doing this before parsing might cause the values to break the JSON (e.g. variable value as `\`), it would be better to that after parsing.